src/App.js

```diff
@@ -1,25 +1,37 @@
-import React, {useState} from 'react'; 
+import React, {useState,useEffect} from 'react'; 
 import TodoList from './TodoList'; //remember the ./ because it's in a different file
 import AddTodoForm from './AddTodoForm';
 
-
+const useSemiPersistentState=()=>{ 
+  const [todoList,setTodoList]=useState(JSON.parse(localStorage.getItem('savedTodoList')))
```

Comment:
This will actually fail if you remove the `savedTodoList` manually from your local storage. Go ahead and try it and see what happens!

The solution here is to do something like this
`JSON.parse(localStorage.getItem('savedTodoList')) ?? []`

the `??` operator is this https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator 
Pretty much what it does is it takes the left side when the right side is null or undefined.